### ISSUE TITLE
max_shown_comments: has no effect

### ISSUE THREAD

**Issue by Shinobi32** [umputun/remark42#812]
When I change max_shown_comments: to 5 it has no effect.
I can change the theme to light/dark but not max_shown_comments.

Maybe I misunderstand the function of max_shown_comments:. 
Is it supposed to limit the number of comments shown on a page before pagination or scrollbar kicks in?

**Comment by bdkbdk123** [comment]
Afaik it only works in mobile view (when your client sends appropriate headers). In order to make it work in desktop view I used a dirty workaround and in `frontend/app/common/constants.ts` replaced
`export const IS_MOBILE = /Android|webOS|iPhone|iPad|iPod|Opera Mini|Windows Phone/i.test(navigator.userAgent);`
with
`export const IS_MOBILE = true;`

You will have to build image from source after this change

**Comment by Shinobi32** [comment]
Thanks for the workaround :)

**Comment by akellbl4** [comment]
@umputun wdyt? I don't know why this option works only for mobile. Should we do the same for desktop?

**Comment by umputun** [comment]
as far as I recall we used to have that options on both but decided to drop on desktop. Something didn't work right with this, don't recall what exactly, maybe google indexing or navigation to the comment? 

probably there is an old ticket about it somewhere. If we enable it I'd suggest making it optional because such pagination on a desktop is not something one usually need. I don't recall what max_shown_comments value for unlimited is (0 ?) but I think whatever it is we should have it as a default.

**Comment by Shinobi32** [comment]
I disagree that the number of comments shown doesn't usually matter on desktop. What if you have a comments thread with hundreds or thousands of comments? Endless scrolling is not a good look IMO.

Most national Newspapers (UK) for example show a few comments on page load and then have a "click to load more comments" button. I think that's probably a better way to go than pagination because that way the comments would be easier for Google to index them.

**Comment by paskal** [comment]
Still reproduces on master, and the setting does reach the widget, so the value is being read and then ignored rather than lost on the way in.

Four root comments on one thread, embedded with the setting at two:

```js
window.remark_config = { host, site_id: 'remark', url: thread, max_shown_comments: 2 }
```

The iframe `src` carries `max_shown_comments=2`, and the widget document renders four `article` elements. Checked against the end-to-end stack while writing the configuration cases in #2196; no test for it is in that PR, since it would be red.

The shape the test should take once this is fixed: four seeded comments, the setting at two, a count of `article` elements, and a control that reveals the rest.
